docs/tutorials/react-native.rst

```diff
@@ -0,0 +1,141 @@
+
+If you're looking for a working solution, there is a `demo app available via Expo <https://exp.host/@vonovak/js-lingui-demo>`_ and its source code is `here <https://github.com/vonovak/js-lingui-demo>`_.
+
+Let's Start
+==============
+
+We're going to translate the following app:
+
+.. code-block:: jsx
+
+ import React from 'react';
+ import { StyleSheet, Text, View, Alert, SafeAreaView, Button } from 'react-native';
+ 
+ export default class App extends React.Component {
+   render() {
+     return (
+       <Inbox
+         user={{
+           lastLogin: 'today',
+         }}
+         markAsRead={this.showAlert}
+         messages={[]}
+       />
+     );
+   }
+ 
+   showAlert = () => {
+     Alert.alert('', 'Do you want to set all your messages as read?');
+   };
+ }
+ 
+ const Inbox = ({ messages, markAsRead, user }) => {
+   const messagesCount = messages.length;
+   const { lastLogin } = user;
+ 
+   return (
+     <SafeAreaView style={styles.container}>
+       <View style={styles.container2}>
+         <Text style={styles.heading}>Message Inbox</Text>
+ 
+         <Text>
+           See all unread messages or
+         </Text>
+         <Button onPress={markAsRead} title="mark messages as read" />
+ 
+         <Text>
+           {messagesCount === 1
+             ? `There's {messagesCount} message in your inbox.`
+             : `There're ${messagesCount} messages in your inbox.`}
+         </Text>
+       </View>
+ 
+       <Text>Last login on {lastLogin}.</Text>
+     </SafeAreaView>
+   );
+ };
+ 
+ const styles = StyleSheet.create({
+   container: {
+     paddingTop: 40,
+     flex: 1,
+     backgroundColor: '#fff',
+     alignItems: 'center',
+   },
+   container2: {
+     alignItems: 'center',
+     flex: 1,
+   },
+   heading: {
+     fontSize: 20,
+     fontWeight: 'bold',
+   },
+ });
+
+As you can see, it's a simple mailbox application with only one screen.
+
+Introducing internationalization
+================================
+
+Not surprisingly, this part isn't too different from the React tutorial.
+
+Let's use the ``Trans`` component first. Don't forget that we need to wrap our root component with the ``I18nProvider`` for ``Trans`` to work correctly.
+
+Let's translate the screen heading:
+
+.. code-block:: jsx
+
+ import { Trans } from '@lingui/react';
+
+ <Text style={styles.heading}><Trans>Message Inbox</Trans></Text>
+
+
+This was easy. Now, the next step is to translate the ``title`` prop of the ``Button`` component. But wait a sec, the button expects to receive a ``string``, so we cannot use the ``<Trans>`` component here! Also notice the same holds also for the ``Alert.alert`` call.
+
+At this point, we need to turn our attention to the ``@lingui/core`` package, namely the ``setupI18n`` method which can give us an object that we can use like this: ``i18n.t`this will be translated``` and the result of such call is a string. Let's get to it!
```

Comment:
This should be actually done using `withI18n` decorator which injects `i18n` object to props. I can't believe I haven't mentioned in React docs... I have to update it too. The only mention about this (with short example) is in [React API reference](https://lingui.github.io/js-lingui/ref/lingui-react.html#withI18n). After that, everything is the same.

Reply:
> This should be actually done

sorry, I don't exactly know what you have in mind. Can you take a look at the final code [here](This should be actually done) and perhaps make a PR?

Or do you have this in mind:

```
<Button onPress={markAsRead} title={i18n.t`mark messages as read`} />
```

to become 

```
<Button onPress={markAsRead} title={this.props.i18n.t`mark messages as read`} />
``` 
?

My idea is to show both translation in react (`Trans`/ `this.props.i18n`) and outside of react (`i18n.t`)